osquery/logger/plugins/buffered.cpp

```diff
@@ -67,14 +73,53 @@ void BufferedLogForwarder::check() {
       VLOG(1) << "Error sending status to logger: " << status.getMessage();
     } else {
       // Clear the status logs once they were sent.
-      iterate(indexes, ([&results, this](std::string& index) {
+      iterate(indexes, ([this](std::string& index) {
                 if (!isStatusIndex(index)) {
                   return;
                 }
                 deleteDatabaseValue(kLogs, index);
               }));
     }
   }
+
+  // Purge any logs exceeding the max after our send attempt
+  if (FLAGS_buffered_log_max > 0) {
+    purge();
+  }
+}
+
+void BufferedLogForwarder::purge() {
+  std::vector<std::string> indexes;
+  auto status = scanDatabaseKeys(kLogs, indexes, index_name_, 0);
```

Comment:
I wonder if, while buffering, the class could maintain a look-aside count? Then instead of potentially buffering 1M indexes to count the size of a vector, this can just [`std::atoi`](http://en.cppreference.com/w/cpp/string/byte/atoi).